conda_smithy/tests/test_feedstock_io.py

```diff
@@ -0,0 +1,710 @@
+        filename1 = "test1.txt"
+        filename2 = "test2.txt"
+
+        write_text = "text"
+        with open(filename1, "w") as fh:
+            fh.write(write_text)
+
+        self.assertTrue(os.path.exists(filename1))
+        self.assertFalse(os.path.exists(filename2))
+
+        fio.copy_file(filename1, filename2)
+
+        self.assertTrue(os.path.exists(filename1))
+        self.assertTrue(os.path.exists(filename2))
+
+        read_text = ""
+        with open(filename2, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual(write_text, read_text)
+
+
+    def tearDown(self):
+        os.chdir(self.old_dir)
+        del self.old_dir
+
+        shutil.rmtree(self.tmp_dir)
+        del self.tmp_dir
+
+
+class TestFeedstockIO_w_Git(unittest.TestCase):
+    def setUp(self):
+        self.old_dir = os.getcwd()
+
+        self.tmp_dir = tempfile.mkdtemp()
+        self.repo = git.Repo.init(self.tmp_dir)
+        os.chdir(self.tmp_dir)
+
+        with open(os.path.abspath(".keep"), "w") as fh:
+            fh.write("")
+
+
+    def test_repo_1(self):
+        self.assertTrue(isinstance(fio.get_repo(self.tmp_dir), git.Repo))
+
+
+    def test_repo_2(self):
+        self.assertTrue(isinstance(fio.get_repo(""), git.Repo))
+
+
+    def test_get_file_blob_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        self.repo.index.add([filename])
+
+        blob = None
+        try:
+            blob = fio.get_file_blob(self.repo, filename)
+        except StopIteration:
+            self.fail("Unable to find the file we added.")
+
+        self.assertEqual(blob.name, os.path.basename(filename))
+
+
+    def test_get_file_blob_2(self):
+        filename = "test.txt"
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        self.repo.index.add([filename])
+
+        blob = None
+        try:
+            blob = fio.get_file_blob(self.repo, filename)
+        except StopIteration:
+            self.fail("Unable to find the file we added.")
+
+        self.assertEqual(blob.name, os.path.basename(filename))
+
+
+    def test_get_mode_file_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        perms = [
+            stat.S_IWUSR,
+            stat.S_IXUSR,
+            stat.S_IRUSR,
+            stat.S_IXGRP,
+            stat.S_IRGRP,
+            stat.S_IROTH
+        ]
+
+        set_mode = functools.reduce(op.or_, perms)
+
+        blob = self.repo.index.add([filename])[0].to_blob(self.repo)
+        blob.mode = set_mode
+        self.repo.index.add([blob])
+
+        os.chmod(filename, set_mode)
+
+        file_mode = fio.get_mode_file(filename)
+
+        self.assertEqual(file_mode & set_mode, set_mode)
+
+
+    def test_get_mode_file_2(self):
```

Comment:
`test_get_mode_file_1` and `test_get_mode_file_2` are basically the same since `os.chdir(self.tmp_dir)` is done during `setUp`, so can we remove one of them?